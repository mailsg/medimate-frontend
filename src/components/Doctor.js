import React from 'react';
import {
  FaAngleRight,
} from 'react-icons/fa';
import { GoGear } from 'react-icons/go';
import styles from '../css/doctor.module.css';
import image1 from '../assets/doc-details.png';

export const Doctor = () => (
  <section>
    <article className={styles.photo}>
      <img className={styles['doctor-img']} src={image1} alt="laura" />
    </article>
    <article className={styles['extreme-right']}>
      <header>
        <h1>Laura Bush</h1>
        <p className={styles['small-fonts']}>A few words description</p>
      </header>
      <div className={styles['attribute-ls']}>
        <div className={[styles.attribute, styles['dark-bg']].join(' ')}>
          <p>Fee per Appointment</p>
          <p>$145</p>
        </div>
        <div className={[styles.attribute, styles['light-bg']].join(' ')}>
          <p>Specialization</p>
          <p>Physiotherapy</p>
        </div>
        <div className={[styles.attribute, styles['dark-bg']].join(' ')}>
          <p>Time available</p>
          <p>8am - 4pm</p>
        </div>
        <div className={[styles.attribute, styles['light-bg']].join(' ')}>
          <p>Duration</p>
          <p>2 hours</p>
        </div>
      </div>
      <p className={styles.discount}>
        <strong>5.9% APR</strong>
        Repesentative
      </p>
      <p className={styles['small-fonts']}>DISCOVER MORE DOCTORS</p>
      {/* <img src={image1} alt="laura" /> */}
      <button type="button">
        <GoGear />
        Reserve
        <FaAngleRight />
      </button>
    </article>
  </section>
);

export default Doctor;
