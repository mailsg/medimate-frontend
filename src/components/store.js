import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './redux/doctorSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
  },
});

export default store;
