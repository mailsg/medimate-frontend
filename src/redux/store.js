import { configureStore } from '@reduxjs/toolkit';
import AppointmentSlice from './appointmentSlice';
import DoctorSlice from './doctorSlice';
import appReducer from './adddoctorSlice';

const store = configureStore({
  reducer: {
    doctor: DoctorSlice,
    appointment: AppointmentSlice,
    app: appReducer,
  },
});

export default store;
