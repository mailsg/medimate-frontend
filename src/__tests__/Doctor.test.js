import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Doctor from '../components/Doctor';
import store from '../redux/store';

describe('Doctor Profile Component', () => {
  test('renders Doctor Profile component', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Doctor />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
