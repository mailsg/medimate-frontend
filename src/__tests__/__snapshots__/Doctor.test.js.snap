// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor Profile Component renders Doctor Profile component 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
